### PR TITLE
rename STOI

### PR DESCRIPTION
**Pull request by Borda**
## What does this PR do?

unify the STOI name, part of #729 

## Before submitting

- [x] Was this discussed/approved via a Github issue? (no need for typos and docs improvements)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), Pull Request section?
- [x] Did you make sure to update the docs?
- [x] Did you write any new necessary tests?

## PR review

Anyone in the community is free to review the PR once the tests have passed.
If we didn't discuss your PR in Github issues there's a high chance it will not be merged.

## Did you have fun?

Make sure you had fun coding 🙃

### CODE REVIEW

**Review by ethanwharris — APPROVED** [review]

File: torchmetrics/audio/stoi.py [hunk 1]
```diff
@@ -131,3 +133,32 @@ def update(self, preds: Tensor, target: Tensor) -> None:  # type: ignore
     def compute(self) -> Tensor:
         """Computes average STOI."""
         return self.sum_stoi / self.total
+
+
+class STOI(ShortTermObjectiveIntelligibility):
```
Comment (rohitgr7) [inline torchmetrics/audio/stoi.py:138]: are there any deprecation tests?

**Review by rohitgr7 — APPROVED** [review]

File: torchmetrics/audio/stoi.py (hunk 1 above)
Comment (Borda) [reply torchmetrics/audio/stoi.py:138]: not for raising warning, just for the outputs, see https://github.com/PyTorchLightning/metrics/pull/752#discussion_r784751844

**Review by SkafteNicki — APPROVED** [review]

**Comment by quancs** [comment]
@Borda I find STOI is more offen called **Short-Time Objective Intelligibility** than **Short-Term Objective Intelligibility**. Could we change the full name to **Short-Time Objective Intelligibility**?

**Comment by quancs** [comment]
Seems its name in papers is **Short-Time Objective Intelligibility** (a mistake in older doc :) ). So, I guess it's better to change it before a new release

**Pull request merged**